node/src/ledger/mod.rs

```diff
@@ -27,3 +28,100 @@ lazy_static::lazy_static! {
 pub fn ledger_empty_hash_at_depth(depth: usize) -> LedgerHash {
     LEDGER_HASH_EMPTIES.get(depth).unwrap().clone()
 }
+
+/// Given the hash of the subtree containing all accounts of height `subtree_height`
+/// compute the hash of a tree of size `LEDGER_DEPTH` if all other nodes were
+/// empty.
+pub fn complete_height_tree_with_empties(
+    content_hash: &LedgerHash,
+    subtree_height: usize,
+) -> LedgerHash {
+    assert!(LEDGER_DEPTH >= subtree_height);
+    let content_hash = content_hash.0.to_field();
+
+    let computed_hash = (subtree_height..LEDGER_DEPTH).fold(content_hash, |prev_hash, height| {
+        let empty_right = ledger::V2::empty_hash_at_height(height);
```

Comment:
`ledger::V2::empty_hash_at_height` is quite slow, you can use `HASH_EMPTIES` instead:
https://github.com/openmina/openmina/blob/adf6f8129b1b0df513b0083526686bd1b16f3e09/ledger/src/tree.rs#L199

If you could also change it to be `Lazy<Arc<[Fp; 36]>>`, not sure why I've used a mutex there

Reply:
@sebastiencs good catch. Just pushed the version that uses the cached values